patches/minecraft/net/minecraft/server/PlayerAdvancements.java.patch

```diff
@@ -20,11 +20,26 @@
        boolean flag = false;
        AdvancementProgress advancementprogress = this.m_135996_(p_135989_);
        boolean flag1 = advancementprogress.m_8193_();
-@@ -251,6 +_,7 @@
+@@ -246,11 +_,14 @@
+          this.m_136008_(p_135989_);
+          this.f_135967_.add(p_135989_);
+          flag = true;
++         net.minecraftforge.event.ForgeEventFactory.onAdvancementProgressedEvent(this.f_135968_, p_135989_, advancementprogress, p_135990_, net.minecraftforge.event.entity.player.AdvancementEvents.AdvancementProgressEvent.ProgressType.GRANT);
+          if (!flag1 && advancementprogress.m_8193_()) {
+             p_135989_.m_138321_().m_9989_(this.f_135968_);
              if (p_135989_.m_138320_() != null && p_135989_.m_138320_().m_14996_() && this.f_135968_.f_19853_.m_46469_().m_46207_(GameRules.f_46153_)) {
                 this.f_135962_.m_11264_(new TranslatableComponent("chat.type.advancement." + p_135989_.m_138320_().m_14992_().m_15548_(), this.f_135968_.m_5446_(), p_135989_.m_138330_()), ChatType.SYSTEM, Util.f_137441_);
              }
-+            net.minecraftforge.common.ForgeHooks.onAdvancement(this.f_135968_, p_135989_);
++            net.minecraftforge.common.ForgeHooks.onAdvancement(this.f_135968_,p_135989_);//todo:remove in 1.19
```

Comment:
Formatting: space after comma in method parameters list, and some additional spaces around and in the comment:
```suggestion
+            net.minecraftforge.common.ForgeHooks.onAdvancement(this.f_135968_, p_135989_); // todo: remove in 1.19
```